packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -563,13 +569,17 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   }
 
   void _updateSelectedIndex() {
-    assert(widget.value == null ||
-      widget.items.where((DropdownMenuItem<T> item) => item.value == widget.value).length == 1);
-    _selectedIndex = null;
-    for (int itemIndex = 0; itemIndex < widget.items.length; itemIndex++) {
-      if (widget.items[itemIndex].value == widget.value) {
-        _selectedIndex = itemIndex;
-        return;
+    if (widget.items != null) {
+      assert(widget.value == null ||
+          widget.items
```

Comment:
nit: indent this a bit more so it's vertically aligned (yes, the original code was bad for this too, oops)